sympy/calculus/util.py

```diff
@@ -410,6 +415,30 @@ def periodicity(f, symbol, check=False):
     >>> periodicity(sin(4*x)**cos(2*x), x)
     pi
     >>> periodicity(exp(x), x)
+
+    >>> class F(Function):
+    ...     nargs = 1
+    ...     def _period(self, general_period, symbol):  # This emulates trigonometric function's method.
+    ...         arg = self.args[0]
```

Comment:
This example is much too long for a doctest and is overly complicated for a normal test as well.

Reply:
Made it more simplified.